src/context/LogContext/LogContext.test.tsx

```diff
@@ -16,11 +17,30 @@ describe("useLogContext", () => {
     );
     const { result } = renderHook(() => useLogContext(), { wrapper });
     act(() => {
-      result.current.ingestLines(["foo", "bar"]);
+      result.current.ingestLines(["foo", "bar"], LogTypes.EVERGREEN_TASK_LOGS);
     });
     expect(result.current.logLines).toStrictEqual(["foo", "bar"]);
     expect(result.current.lineCount).toBe(2);
   });
+  it("ingesting a resmoke log should transform it before adding it to the list of logs", () => {
+    const wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
+      <LogContextProvider>{children}</LogContextProvider>
+    );
+    const { result } = renderHook(() => useLogContext(), { wrapper });
+    act(() => {
+      result.current.ingestLines(
+        [
+          `[j0:s0:n1] {"t":{"$date":"2022-09-13T16:57:46.852+00:00"},"s":"D2", "c":"REPL_HB",  "id":4615670, "ctx":"ReplCoord-1","msg":"Sending heartbeat","attr":{"requestId":3705,"target":"localhost:20003","heartbeatObj":{"replSetHeartbeat":"shard-rs0","configVersion":5,"configTerm":3,"hbv":1,"from":"localhost:20004","fromId":1,"term":3,"primaryId":1}}}`,
+          `[j0:s0] {"t":{"$date":"2022-09-13T16:57:46.855+00:00"},"s":"I",  "c":"-",        "id":20883,   "ctx":"conn188","msg":"Interrupted operation as its client disconnected","attr":{"opId":6047}}`,
+        ],
+        LogTypes.RESMOKE_LOGS
+      );
+    });
+    expect(result.current.logLines).toStrictEqual([
+      `[j0:s0:n1] | 2022-09-13T16:57:46.852+00:00 D2 REPL_HB  4615670 [ReplCoord-1] "Sending heartbeat","attr":{"requestId":3705,"target":"localhost:20003","heartbeatObj":{"replSetHeartbeat":"shard-rs0","configVersion":5,"configTerm":3,"hbv":1,"from":"localhost:20004","fromId":1,"term":3,"primaryId":1}}`,
+      `[j0:s0] | 2022-09-13T16:57:46.855+00:00 I  -        20883   [conn188] "Interrupted operation as its client disconnected","attr":{"opId":6047}`,
```

Comment:
I wonder if there is a way we can test this without the exact string whitespaces... Maybe we can trim and compare? it makes me kind of sad that we'd need to update many places if we ever decide to change the `padEnd` values 😭

Reply:
I think the whitespaces are part of the log. I know server has some regexp that they use to find errors so i fear changing the tests would break things.